pkg/secrets/manager.go

```diff
@@ -33,3 +35,24 @@ type Manager interface {
 	// deployment, or an error if one can not be constructed.
 	Decrypter() (config.Decrypter, error)
 }
+
+// AreCompatible returns true if the two Managers are of the same type and have the same state.
+func AreCompatible(a, b Manager) bool {
+	if a == nil || b == nil {
+		return a == nil && b == nil
+	}
+
+	if a.Type() != b.Type() {
+		return false
+	}
+
+	as, err := json.Marshal(a.State())
```

Comment:
I assume this is fine, but just to point it out, perhaps we should be using `reflect.DeepEquals` instead? Since that is specifically designed to do this type of comparison, whereas it's possible that the JSON encoding of two distinct types might be identical?
https://golang.org/pkg/reflect/#DeepEqual

Reply:
I chose this implementation (even though I find it a bit distasteful) because the contract for `secrets.Manager` requires that the value returned by `State()` can be marshalled to and from JSON, and this particular implementation allows for the `State` values to be of distinct types.